source/material_model/utilities.cc

```diff
@@ -779,6 +775,13 @@ namespace aspect
       {
         const unsigned int n_fields = masses.size();
         std::vector<double> volumes(n_fields);
+
+        if (n_fields == 1)
+          {
+            volumes[0] = 1.0;
+            return volumes;
+          }
+
```

Comment:
Shouldnt the existing code below do the same thing for n_fields == 1? If so remove this to keep the function simpler (also: with your current implementation what would happen if n_fields == 1, but return_as_fraction == false? then this function would return the wrong value I think).

Reply:
No, I actually implemented this because a test failed. The existing version of this assumed that there is always a background field, so it didn't run into this problem. But there were steinberger tests without a background field, that still had compositions of zero somewhere. In that case, sum_volumes is zero and you get a floating point exception. But I can fix the case with return_as_fraction == false.